automation-ui/backoffice/src/test/java/com/yas/automation/ui/pages/ProductPage.java

```diff
@@ -226,7 +229,11 @@ public List<WebElement> getAllRowsInTable() {
     }
 
     public void clickToEditProductBtn() {
-        this.wait(Duration.ofSeconds(1));
+        Supplier<Boolean> checkData = () -> {
+            return getAllRowsInTable().isEmpty();
```

Comment:
We should simplify it in lambda style `this.waitWithRetry(Duration.ofSeconds(1), () -> getAllRowsInTable().isEmpty());`